src/main/java/yapion/serializing/YAPIONSerializer.java

```diff
@@ -64,7 +64,11 @@ public YAPIONSerializer(@NonNull Object object, String context) {
 
     private YAPIONSerializer(@NonNull Object object, YAPIONSerializer yapionSerializer) {
         this.object = object;
-        this.contextManager = yapionSerializer.contextManager;
+        if (yapionSerializer.contextManager.isCascading()) {
+            this.contextManager = yapionSerializer.contextManager;
+        } else {
+            this.contextManager = new ContextManager(yapionSerializer.contextManager.get());
+        }
```

Comment:
Is this for handling the YAPIONData.cascading property correctly?

Reply:
Basically, yes. 

If cascading is not used directly on the root object, but a little deeper in an object tree, the YAPIONData annotation with cascading should only be applied to deeper objects and not to parent objects. Therefore you cannot use the same context manager on all levels, sometimes you need a fresh one.

You will find the same code change in YAPIONDeserializer starting at line 154.

Reply:
Thanks for the explanation.